server/src/main/java/com/broadcom/lsp/cobol/service/CobolLanguageServer.java

```diff
@@ -126,12 +134,45 @@ private void getLocaleFromClient() {
 
   @Override
   public CompletableFuture<Object> shutdown() {
-    return supplyAsync(() -> TRUE);
+    LOG.info("LS for COBOL received shutdown request");
+    try {
+      cancelAllProcessing();
+      exitCode = 0;
+    } catch (Exception exception) {
+      return CompletableFuture.completedFuture(new ShutdownResponse(null, exception.getMessage()));
+    }
+    return CompletableFuture.completedFuture(new ShutdownResponse(null, String.valueOf(exitCode)));
+  }
+
+  /**
+   * Attempts to stop all actively executing tasks, halts the processing of waiting tasks, and
+   * returns a list of the tasks that were awaiting execution. These tasks are drained (removed)
+   * from the task queue upon return from this method.
+   *
+   * <p>There are no guarantees beyond best-effort attempts to stop processing actively executing
+   * tasks. This implementation cancels tasks via Thread.interrupt(), so any task that fails to
+   * respond to interrupts may never terminate.
+   *
+   * <p>Source:
+   * https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html#shutdownNow--
+   */
+  private void cancelAllProcessing() {
+    // NOTE: shutdownNow principally is not a bad practice when we know what we are doing.
+    //      Here in this case, since we are only stopping our own threads which don't
+    //      require any clean up.It's ok. Java doc doesn't mention any bad practice for this.
+
+    // cancels all the running task and clears empty queue on forkJoinPool.
+    ForkJoinPool.commonPool().shutdownNow();
+
+    // cancels all the running task and clears empty queue on CobolTextDocumentService executor.
+    ((CobolTextDocumentService) getTextDocumentService()).disposeAllOperation(this);
+    LOG.info("All processing abandoned as per shutdown call");
   }
 
   @Override
   public void exit() {
-    // not supported
+    LOG.info("LS for COBOL exiting with code: " + exitCode);
```

Comment:
I just tested it and noticed that this message should be "COBOL LS server exits with code:"

Reply:
Done